client/my-sites/sidebar/sidebar.jsx

```diff
@@ -146,21 +147,14 @@ export class MySitesSidebar extends Component {
 			return null;
 		}
 
-		let sparkline;
-		if ( ! site.jetpack && ! this.isItemLinkSelected( 'stats' ) ) {
-			sparkline = (
-				<img
-					className="sidebar__sparkline"
-					src={ `https://${ site.wpcom_url }/wp-includes/charts/admin-bar-hours-scale-2x.php?masterbar=1` } />
-			);
-		}
+		const siteId = site ? site.ID : null;
 
 		return (
 			<li className={ this.itemLinkClass( '/stats', 'stats' ) }>
 				<SiteStatsStickyLink onClick={ this.onNavigate }>
 					<Gridicon icon="stats-alt" size={ 24 } />
 					<span className="menu-link-text">{ this.props.translate( 'Stats' ) }</span>
-					{ sparkline }
+					<StatsSparkline className="sidebar__sparkline" siteId={ siteId } />
```

Comment:
Dunno how you feel about its readability, but I might have opted for pulling in `_.get` to avoid the ternary and addition variable:

```js
siteId={ get( site, 'ID' ) }
```